### PR TITLE
[SYCLomatic] Add support for handling CU_EVENT_* enum

### PR DESCRIPTION
**Pull request by the-slow-one**
cuEventCreate can create events based on how host thread behaves while waiting for the event to finish. 
Signed-off-by: [Deepak Raj H R](Deepak.raj.h.r@intel.com)

### CODE REVIEW

File: clang/lib/DPCT/RulesLang/RulesLang.cpp [hunk 1]
```diff
@@ -2879,6 +2883,15 @@ bool EventAPICallRule::isEventElapsedTimeFollowed(const CallExpr *Expr) {
 }
 
 void EventAPICallRule::runRule(const MatchFinder::MatchResult &Result) {
+  if (auto *DRE = getNodeAsType<DeclRefExpr>(Result, "eventEnum")) {
+    if (auto *EC = dyn_cast<EnumConstantDecl>(DRE->getDecl())) {
+      std::string EName = EC->getName().str();
+      report(DRE->getBeginLoc(), Diagnostics::UNSUPPORTED_EVENT_FEATURE, false,
+             EName);
+      emplaceTransformation(new ReplaceStmt(DRE, "0x0"));
+    }
```
Comment (ziranzha) [inline clang/lib/DPCT/RulesLang/RulesLang.cpp:2892]: Should return after migration.

File: clang/lib/DPCT/RulesLang/RulesLang.cpp [hunk 2]
```diff
@@ -2879,6 +2883,15 @@ bool EventAPICallRule::isEventElapsedTimeFollowed(const CallExpr *Expr) {
 }
 
 void EventAPICallRule::runRule(const MatchFinder::MatchResult &Result) {
+  if (auto *DRE = getNodeAsType<DeclRefExpr>(Result, "eventEnum")) {
+    if (auto *EC = dyn_cast<EnumConstantDecl>(DRE->getDecl())) {
+      std::string EName = EC->getName().str();
+      report(DRE->getBeginLoc(), Diagnostics::UNSUPPORTED_EVENT_FEATURE, false,
+             EName);
+      emplaceTransformation(new ReplaceStmt(DRE, "0x0"));
```
Comment (ziranzha) [inline clang/lib/DPCT/RulesLang/RulesLang.cpp]: How about using `0` instead of `0x0`?

**Review by ziranzha — APPROVED** [review]

**Review by TejaX-Alaghari — APPROVED** [review]
LGTM

File: clang/lib/DPCT/Diagnostics/Diagnostics.inc [hunk 3]
```diff
@@ -296,6 +296,8 @@ DEF_WARNING(SPARSE_NNZ, 1134, MEDIUM_LEVEL, "The tool cannot deduce the consumer
 DEF_COMMENT(SPARSE_NNZ, 1134, MEDIUM_LEVEL, "The tool cannot deduce the consumer API (\"dpct::sparse::csrgemm\") of this API, and this API has 2 arguments depending on the 8th and the 12th parameters of the consumer API. Please replace the 2 arguments tagged as \"dpct_placeholder\" with the corresponding value.")
 DEF_WARNING(JOINT_MATRIX_SHAPE, 1135, HIGH_LEVEL, "Please check if joint_matrix implementations support the combination of data type and matrix shape type in the target hardware.")
 DEF_COMMENT(JOINT_MATRIX_SHAPE, 1135, HIGH_LEVEL, "Please check if joint_matrix implementations support the combination of data type and matrix shape type in the target hardware.")
+DEF_WARNING(UNSUPPORTED_EVENT_FEATURE, 1136, LOW_LEVEL, "A SYCL event does not support the feature of a CUDA event enabled with flag \"%0\".")
```
Comment (zhimingwang36) [inline clang/lib/DPCT/Diagnostics/Diagnostics.inc]: try to reuse 1014 .   (
The {.*} are not supported for SYCL {.*}. The output parameter(s) are set to 0."

Comment (the-slow-one) [reply clang/lib/DPCT/Diagnostics/Diagnostics.inc]: Done

**Review by zhimingwang36 — APPROVED** [review]

**Pull request merged**